tests/unit/lib/tasks/TaskScheduler.spec.js

```diff
@@ -115,4 +116,94 @@ describe('TaskScheduler', () => {
             }
         }
     });
+
```

Comment:
can you please double-check we are well covered on this class (including pre-existing code)?

4 tests does not seem like a lot, esp. seeing all the branches in the code (and the subtelties that can arise)

Reply:
The test i added goes through most of the paths possible but i added unit tests for each of the methods.